app/javascript/components/student/mentoring-session/mentoring-request/MentoringRequestForm.tsx

```diff
@@ -63,6 +64,15 @@ export const MentoringRequestForm = ({
     }
   )
 
+  const handleSubmit = useCallback(
+    (e) => {
+      e.preventDefault()
+
+      mutation()
+    },
+    [mutation]
+  )
+
```

Comment:
The page was reloading on every submission because we weren't called `e.preventDefault()`